contracts/cw-proposal-multiple/src/contract.rs

```diff
@@ -696,16 +713,25 @@ pub fn query_info(deps: Deps) -> StdResult<Binary> {
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
-    if msg.id % 2 == 0 {
-        // Proposal hook so we can just divide by two for index
-        let idx = msg.id / 2;
-        PROPOSAL_HOOKS.remove_hook_by_index(deps.storage, idx)?;
-    } else {
-        // Vote hook so we can minus one then divid by two for index
-        let idx = (msg.id - 1) / 2;
-        VOTE_HOOKS.remove_hook_by_index(deps.storage, idx)?;
+    let repl = TaggedReplyId::new(msg.id);
```

Comment:
LOVE this abstaction. Great work!

Reply:
Thanks, was thinking if I use `#[inline]` attribute for `new()`  it can optimize it into single match? _But i could be wrong here_ .Now it doesn't matter since we return `Result` now 😆.